src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs

```diff
@@ -156,8 +156,9 @@ public bool Equals(DiagnosticData? other)
                 return false;
             }
 
+            // TODO: unclear why we're only looking at the OriginalFileSpan of the location, and only the start point of it.
```

Comment:
these might need a tracking issue since this code will be around even after we delete the old impl